mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -795,8 +799,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       require(a.length == k)
       copyToDouble(a)
       blas.dspr(upper, k, c, da, 1, ata)
-      if (b != 0.0) {
-        blas.daxpy(k, c * b, da, 1, atb, 1)
+      if (Math.abs(b) > Double.MinPositiveValue) {
```

Comment:
You are right, I should pick a looser threshold. It seems that this check is only really to prevent extra work, since daxpy will just be adding a zeros vector if b==0.